### ISSUE TITLE
Haxe 4-rc.3 Cppia: Declaring a static function with dynamic parameter fails to compile.

### ISSUE THREAD

**Issue by skylerparr** [HaxeFoundation/hxcpp#837]
Using hxcpp 4.0.19. When trying to compile
```
class EitherSupport {
  public static function getValue(e: Dynamic): Void {

  }
}
```
I get a compilation error on cppia 
```
Error reading file Bad class type, line 14, char 9
```
Here's the generated cppia code:
```
CPPIB
5
0 
3 new
4 main
8 getValue
1 e
5
0 
8 AnnaLang
4 void
13 EitherSupport
7 Dynamic
2
D		D��	KM
```
Am I missing something?

**Comment by skylerparr** [comment]
@hughsando This looks like it's still an issue with haxe 4.2.4 and hxcpp 4.2.1. I've narrowed it down to getting this error when passing arguments to a function. This may be isolated to just windows. I wouldn't mind tracking down the issue and creating a PR if you give me a some tips on how to track this down.

**Comment by hughsando** [comment]
If you compile with `-debug -D annotate-source` the cppia file will have more information that might help. It might also avoid the problem by using ascii out, but that would be some more information too.

**Comment by skylerparr** [comment]
I just re-read my original post.  I'm sorry, this is not a compilation error. This is the error coming out of `Module.fromString(cppiaCode);` and the other functions in `Module` at runtime.  I just tried adding the `-D annotate-source` to my compilation flags and the entire problem went away. Problem solved?!!?

**Comment by hughsando** [comment]
Sounds like there is some incompatibility in the binary format.  With just
debug (not  annotate-source)do you still get the problem?
It could either be the binary write, binary reader or maybe some unexpected
character in the file.
The binary and ascii version should parallel each other quite closely, so
you might be able to work out what it is trying to read by comparing the
two.

**Comment by skylerparr** [comment]
`-D annotate-source` worked regardless of the `-debug` flag. 
re:
`The binary and ascii version should parallel each other quite closely`
I'm not quite following what you mean there. Are you talking about looking at the generated cppia file? Is there a way to compile to binary vs ascii?

**Comment by hughsando** [comment]
Yes, the debug version is Ascii (Starts with CPPIA) and the release version
is binary (starts with "CPPIB")
The ascii version with annotations is almost readable

**Comment by tobil4sk** [comment]
> This is the error coming out of Module.fromString(cppiaCode);

This sounds like it was the problem.

With hxcpp 9d6db960775e7f2eb75014e27fd66b0b402431e8 and haxe 4.3.7:
```haxe
// Host.hx
import cpp.cppia.Module;
import sys.io.File;

class Host {
        static public function main():Void {
                trace('Hello from cppia HOST');
                final scriptname = './script.cppia';

                trace('Loading as bytes');
                final module = Module.fromData(File.getBytes(scriptname).getData());
                module.boot();
                trace('Loaded successfully');

                trace('Loading as string');
                final module = Module.fromString(File.getContent(scriptname));
                module.boot();
                trace('Loaded successfully');
        }
}
```
```haxe
// Client.hx
class EitherSupport {
        public static function getValue(e: Dynamic): Void {}
}

function main() {}
```

```shell
$ haxe -m Host --cpp bin -D scriptable -D dll_export=classes.info
# ...
$ haxe -m Client --cppia script.cppia -D dll_import=classes.info
# ...
$ bin/Host
Host.hx:6: Hello from cppia HOST
Host.hx:9: Loading as bytes
Host.hx:11: Booting
Host.hx:13: Loaded successfully
Host.hx:15: Loading as string
Error : Error reading file Bad class type, line 13, char 9
```

The problem is that loading the script into a string results in an attempted utf-8 to utf-16 conversion, but the binary cppia format is going to contain bytes that are invalid when interpreted as utf-8. This results in the binary data being corrupted, and therefore the loader fails to read the script data. The correct way to load the script is as Bytes, and then using `Module.fromData`, as demonstrated in the sample. The reason the ascii version works is because that is a text format which avoids the corruption.
